cli/flox-rust-sdk/src/models/environment/remote_environment.rs

```diff
@@ -79,20 +86,35 @@ impl RemoteEnvironment {
         )
         .unwrap();
 
-        let out_link = gcroots_dir(flox, &pointer.owner).join(remote_branch_name(&pointer));
+        let inner_out_link = gcroots_dir(flox, &pointer.owner).join(remote_branch_name(&pointer));
+        let mut inner =
+            ManagedEnvironment::open_with(floxmeta, flox, pointer, dot_flox_path, inner_out_link)
+                .map_err(RemoteEnvironmentError::OpenManagedEnvironment)?;
 
-        let inner = ManagedEnvironment::open_with(floxmeta, flox, pointer, dot_flox_path, out_link)
-            .map_err(RemoteEnvironmentError::OpenManagedEnvironment)?;
+        inner
+            .pull(true)
+            .map_err(RemoteEnvironmentError::ResetManagedEnvironment)?;
+
+        let out_link = {
+            let link_path = path.join(GCROOTS_DIR_NAME);
+            Self::update_out_link(flox, &link_path, &mut inner)?;
+            link_path
+        };
 
-        Ok(Self { inner })
+        Ok(Self { inner, out_link })
     }
 
-    /// Pull a remote environment into a temporary managed environment
+    /// Pull a remote environment into a flox-provided managed environment
+    /// in `<FLOX_CACHE_DIR>/remote/<owner>/<name>`
```

Comment:
suggestion nonblocking: explain the overall layout a bit more
This directory contains env.json and env.lock, as well as a run directory.
There's what we call an "inner out link" in the run directory
The manifest is still stored in git, and there's a second gcroot we keep elsewhere

note:
Not to change here, and I've lost track of some of the details, so maybe all the complexity is necessary. But I'm feeling like some of the indirection is overly complex. At least for something like `activate -r`, it seems like we just need to check for new git commits, if there are call `buildenv $(git show XXX)`, and activate with the outlink. I guess part of it's about sharing code with `ManagedEnvironment`, but creating an env.json and env.lock that contain the same info as the git branch and a second out link with mtimes we depend on feels unnecessary.